File: bringmehomeUI/bringmehome_android/lib/models/requests/user_request.dart
class UserRequest {
  final String firstName;
  final String lastName;
  final String email;
  final String username;
  final String password;
  final String phoneNumber;
  final bool isActive;

  UserRequest(
      {required this.firstName,
      required this.lastName,
      required this.email,
      required this.username,
      required this.password,
      required this.phoneNumber,
      this.isActive = true
  });

  Map<String,dynamic>toJson()=> {
    'firstName': firstName,
    'lastName': lastName,
    'email': email,
    'username': username,
    'password': password,
    'phoneNumber': phoneNumber,
    'isActive': isActive,
  };

}

File: BringMeHome/BringMeHome.Models/Requests/UserRequest.cs
﻿using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BringMeHome.Models.Requests
{
    public class UserRequest
    {
        [Required]
        [MaxLength(50)]
        public string FirstName { get; set; } = string.Empty;

        [Required]
        [MaxLength(50)]
        public string? LastName { get; set; } = string.Empty;

        [Required]
        [MaxLength(100)]
        [EmailAddress]
        public string Email { get; set; } = string.Empty;

        [Required]
        [MaxLength(100)]
        public string Username { get; set; } = string.Empty;

        public byte[]? UserImage { get; set; }

        [MaxLength(20)]
        public string? PhoneNumber { get; set; }

        public string? Address { get; set; }

        public int? CityID { get; set; }

        public bool IsActive { get; set; } = true;

        // Only used when creating a new user
        [MinLength(6)]
        public string? Password { get; set; }
    }
}
